docs/source/configuring.md

```diff
@@ -0,0 +1,145 @@
+retained in terminal history.
+
+We don't recommend using this method in automation or CI; We'd recommend using
+environment variables (below).
+
+```
+rover config profile auth
+```
+
+This command will save the key to the `default` profile unless you pass a 
+`--profile` option to it. 
+
+### Environment Variable
+
+> Environment variables take precedence over the `auth` command. This means 
+that if you set an enviroment variable, it will override any configuration
+a previously run `auth` command set.
+
+To use environment variables as an auth token, make sure it's loaded as the
+`APOLLO_KEY` variable.
+
+It's best practice to use a graph scoped key (beginning with `service:`) in CI 
+environments, rather than a user-scoped key (starting with `user:`) key. 
+The difference in the two is that a graph key is scoped to have permissions to
+access only a single graph, whereas user keys can be used to access any graph a 
+user has access to. For CI, it's best to keep keys as specific as possible, and
+only with the scope needed.
+
+> TODO: where to find graph keys
+
+## Anonymous Telemetry
+
+By default, Apollo collects anonymous data about how Rover is being used. Rover 
+doesn't collect any personally identifiable information like api keys, 
+graph names, file paths, etc. 
+
+If you would like to disable telemetry, you can set the 
+`APOLLO_TELEMETRY_DISABLED` env variable to `1`.
+
+For more info on how Apollo collects and uses this data, see 
+[our privacy notice](./privacy).
+
+> TODO: link
+
+## Logging
+
+There are 5 possible levels of logging that can happen in Rover. These include 
+`error`, `warn`, `info`, `debug`, and `trace`, ranging from the quietest logs to
+the most verbose logs. 
+
+By default, only `error`, `warn`, and `info` messages are logged, but any 
+command can be made more or less verbose by changing the log level.
+
+```
+rover graph check my-graph@prod --schema ./schema.graphql --log debug
+```
+
+If logs are unhelpful or unclear, please leave the Rover team feedback in an 
+issue on [GitHub](https://github.com/apollographql/rover/issues)!
+
+## Advanced
+
+### Environment Variables
+
+The `rover` CLI recognizes several environment variables, which better allow
+you to leverage this tool in CI and/or override defaults and pre-configured
+values.
+
+Environment variables have the highest precedence of all configuration. This
+means that the value of an environment value will always override other
+configuration.
+
+| Name                        | Value          | Default    |
+|-----------------------------|----------------|------------|
+| `APOLLO_CONFIG_HOME` | Path to where CLI configuration is stored. | Default OS Configuration directory. |
+| `APOLLO_KEY` | An API Key generated from Apollo Studio. | none | 
+| `APOLLO_TELEMETRY_URL` | The URL where anonymous usage data is reported | [https://install.apollographql.com/telemetry](https://install.apollographql.com/telemetry) |
```

Comment:
probably don't need to document this one tbh, don't see any reason to add this to the public contract.